### DISCUSSION TITLE
Can't assign Public role from a policy

### DISCUSSION THREAD

**Discussion by w0kyj** [directus/directus#24914] [Feature Requests]
### Describe the Bug

When managing a policy, cannot assign the `Public` restricted role to the policy. Must go to the `Public` role and assign the policy from there.

### To Reproduce

1. Create or Edit a "public" policy
1. Try to add a role -- `Public` role is not available in the role selection

<img width="1131" alt="Screenshot 2024-08-08 at 5 09 07 PM" src="https://github.com/user-attachments/assets/f0db79db-c947-414a-b61c-b7c444bd0b74">
<img width="456" alt="Screenshot 2024-08-08 at 3 59 07 PM" src="https://github.com/user-attachments/assets/107ccbfe-f117-452a-a3b9-8bc508ea2c1f">

### Directus Version

v11.0.1

### Hosting Strategy

Directus Cloud

### Database

PostgreSQL

**Comment by pomeh** [comment]
Hello,

I can confirm this behavior (Directus v11.0.2 / Self-Hosted (Docker Image) / SQLite).

When loading studio page "Public policy" (/admin/settings/policies/abf8a154-5b1c-4a46-ac9c-7300570f4f17), a fetch call is done to `GET /roles` endpoint (exactly `/roles?limit=25&fields[]=icon&fields[]=name&fields[]=description&fields[]=id&sort[]=name&page=1&filter[_and][0][$FOLLOW(directus_access,role)][_none][policy][_eq]=abf8a154-5b1c-4a46-ac9c-7300570f4f17`).

In the JSON response, only Administrator role is present. So when we click "Add existing" role, Public role does not show up.

As far as I understand, this problem may be caused by this root issue in REST API https://github.com/directus/directus/issues/23781
